--- flite-1.4-release/configure.in.orig	2016-01-11 22:48:14.124827914 +0000
+++ flite-1.4-release/configure.in	2016-01-11 22:50:22.935344761 +0000
@@ -270,7 +270,7 @@
 # the one I know -- you can still specific --with-audio=alsa
 AC_TRY_COMPILE([#include <alsa/version.h>],
                [int j=
-                #if SND_LIB_SUBMINOR >= 11
+                #if SND_LIB_VERSION >= 0x01000b
                 3;
                 #endif],
               [AUDIODRIVER="alsa"
